### PR TITLE
Add option for relative convenience symlinks

### PR DESCRIPTION
**Pull request by phoenikz**
Extending --experimental_convenience_symlinks with a new option "relative" and setting up convenience link creation to honour that option, if the target path begins with the base path.

### CODE REVIEW

**Comment by haxorz** [comment]
Wow I'm so sorry @phoenikz for completely missing this PR! My fault for not doing anything after Mar 28, 2023. Thank you @meisterT for noticing this and bringing it to my attention.

There are some merge conflicts (due to the code in the main branch changing in the interim, due to my mistaken inaction). I've resolved them myself and will do the merge shortly.

**Comment by haxorz** [comment]
tl;dr - I don't think this code works at all. I'm sorry again for the huge delay due my fault, but I do think we need to revisit https://github.com/bazelbuild/bazel/issues/2351.

-----

@michajlo was reviewing my merge and wisely suggested we have tests for this. I started by reading this PR's diff more carefully and running the code under a debugger. I don't think it works as intended.

```
PathFragment targetForLink =
    target.startsWith(base) && relativeSymlinks
        ? target.relativeTo(base)
        : target.asFragment();
```

There are two issues with this code:

(1) ` target.startsWith(base)` is always going to be `false` in practice, so the new feature doesn't do anything. `target` is going to be a path under the output base, while `base` is going to be the workspace dir (see the callsite of `createLink` in `createOutputDirectoryLinks`).

(2) Even if we did reach the `target.relativeTo(base)` branch, I don't think that does what you want. My interpretation of https://github.com/bazelbuild/bazel/issues/2351 is that you wanted to produce relative symlinks like `../../../blah`. `PathFragment#relativeTo` isn't going to produce a path string with up-level references like that. Check out `FileSystemUtils#relativePath` --- I think that's what you want :)

-----

Since the delay is my fault, I'd like to offer to fix the code and add tests, etc. But can @phoenikz (or someone else) first please confirm my understanding of the FR is correct?

**Comment by meisterT** [comment]
Thank you for proposing this change! We are closing this pull request as the current implementation does not produce relative symlinks as intended (per @haxorz's analysis above), and the branch has merge conflicts after remaining inactive for over two years.
  
Because a working solution likely requires a different approach from scratch, we'll continue tracking any discussion or interest on the underlying feature request in #2351.
